File: TrabajoAcademicoNet/SoftProgModelo/PersonaModelo/Persona.cs
using PUCP.Inf30.SoftProg.Modelo.BaseModelo;
using PUCP.Inf30.SoftProg.Modelo.PersonaModelo.PersonaUtils;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading;
using System.Threading.Tasks;

namespace PUCP.Inf30.SoftProg.Modelo.PersonaModelo
{
    public class Persona : Registro
    {
        public String Dni {  get; set; }
        public String Nombre { get; set; }
        public String ApellidoPaterno { get; set; }
        public String ApellidoMaterno { get; set; }
        public Genero Genero;
        public DateTime FechaNacimiento { get; set; }
        public int Telefono { get; set; }
        
        public Persona() { }
        public Persona(String dni, String nombre, String apellidoPaterno, 
            String apellidoMaterno, Genero genero, DateTime fechaNacimiento, int telefono)
        {
            this.Dni = dni;
            this.Nombre = nombre;
            this.ApellidoPaterno = apellidoPaterno;
            this.ApellidoMaterno = apellidoMaterno;
            this.Genero = genero;
            this.FechaNacimiento = fechaNacimiento;
            this.Telefono = telefono;
        }
        public Persona(String dni, String nombre, String apellidoPaterno, 
            String apellidoMaterno, Genero genero, DateTime fechaNacimiento, int telefono,
            int id, bool activo) : base (id, activo)
        {
            
            this.Dni = dni;
            this.Nombre = nombre;
            this.ApellidoPaterno = apellidoPaterno;
            this.ApellidoMaterno = apellidoMaterno;
            this.Genero = genero;
            this.FechaNacimiento = fechaNacimiento;
            this.Telefono = telefono;
        }

        public String getGeneroString()
        {
            String cadena = "";

            switch (this.Genero)
            {
                case Genero.HOMBRE:
                    cadena = "HOMBRE";
                    break;
                case Genero.MUJER:
                    cadena = "MUJER";
                    break;
                case Genero.NO_ESPECIFICADO:
                    cadena = "NO_ESPECIFICADO";
                    break;
            }

            return cadena;
        }

        public void setGeneroString(String genero)
        {
            switch (genero)
            {
                case "HOMBRE":
                    this.Genero = Genero.HOMBRE;
                    break;
                case "MUJER":
                    this.Genero = Genero.MUJER;
                    break;
                case "NO_ESPECIFICADO":
                    this.Genero = Genero.NO_ESPECIFICADO;
                    break;
            }
        }

    }
    

}

File: Java/TrabajoAcademico/SoftProgModelo/src/main/java/pe/edu/pucp/inf30/softprog/modelo/persona/Persona.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package pe.edu.pucp.inf30.softprog.modelo.persona;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.xml.bind.annotation.XmlTransient;
import jakarta.xml.bind.annotation.adapters.XmlJavaTypeAdapter;
import java.time.LocalDate;
import java.time.LocalDateTime;
import pe.edu.pucp.inf30.softprog.modelo.base.Registro;
import pe.edu.pucp.inf30.softprog.modelo.persona.utils.Genero;
import java.util.Date;
import pe.edu.pucp.inf30.softprog.modelo.venta.utils.LocalDateTimeAdapter;

/**
 *
 * @author Cristhian Horacio
 */
public class Persona extends Registro {

    private String dni;
    private String nombre;
    private String apellidoPaterno;
    private String apellidoMaterno;
    private Genero genero;
    private LocalDateTime fechaNacimiento;
    private int telefono;

    public Persona() {

    }

    public Persona(String dni, String nombre, String apellidoPaterno, String apellidoMaterno, Genero genero, LocalDateTime fechaNacimiento, int telefono) {
        this.dni = dni;
        this.nombre = nombre;
        this.apellidoPaterno = apellidoPaterno;
        this.apellidoMaterno = apellidoMaterno;
        this.genero = genero;
        this.fechaNacimiento = fechaNacimiento;
        this.telefono = telefono;
    }

    public Persona(String dni, String nombre, String apellidoPaterno, String apellidoMaterno, Genero genero, LocalDateTime fechaNacimiento, int telefono, int id, boolean activo) {
        super(id, activo);
        this.dni = dni;
        this.nombre = nombre;
        this.apellidoPaterno = apellidoPaterno;
        this.apellidoMaterno = apellidoMaterno;
        this.genero = genero;
        this.fechaNacimiento = fechaNacimiento;
        this.telefono = telefono;
    }

    public String getDni() {
        return dni;
    }

    public void setDni(String dni) {
        this.dni = dni;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getApellidoPaterno() {
        return apellidoPaterno;
    }

    public void setApellidoPaterno(String apellidoPaterno) {
        this.apellidoPaterno = apellidoPaterno;
    }

    public String getApellidoMaterno() {
        return apellidoMaterno;
    }

    public void setApellidoMaterno(String apellidoMaterno) {
        this.apellidoMaterno = apellidoMaterno;
    }

    public Genero getGenero() {
        return genero;
    }

    public void setGenero(Genero genero) {
        this.genero = genero;
    }

    @XmlJavaTypeAdapter(LocalDateTimeAdapter.class)
    public LocalDateTime getFechaNacimiento() {
        return fechaNacimiento;
    }

    public void setFechaNacimiento(LocalDateTime fechaNacimiento) {
        this.fechaNacimiento = fechaNacimiento;
    }

    public int getTelefono() {
        return telefono;
    }

    public void setTelefono(int telefono) {
        this.telefono = telefono;
    }

    @JsonIgnore
    public String getGeneroString() {
        if (this.genero == null) {
            return null;  // o "" si prefieres cadena vacía
        }
        return this.genero.name(); // "MASCULINO", "FEMENINO", etc.
    }

    public void setGeneroString(String generoStr) {
        if (generoStr == null || generoStr.isEmpty()) {
            this.genero = null;
        } else {
            this.genero = Genero.valueOf(generoStr); // asegúrate que coincide con los nombres del enum
        }
    }

}
